setup.py

```diff
@@ -30,316 +31,110 @@
-        path = os.path.join("cms", "server", "mo", country_code, "LC_MESSAGES")
-        dest_path = os.path.join(USR_ROOT, "share", "locale",
-                                 country_code, "LC_MESSAGES")
-        makedir(dest_path, root, 0755)
-        copyfile(os.path.join(path, "cms.mo"),
-                 os.path.join(dest_path, "cms.mo"),
-                 root, 0644)
-
-    print("creating directories.")
-    dirs = [os.path.join(VAR_ROOT, "log"),
-            os.path.join(VAR_ROOT, "cache"),
-            os.path.join(VAR_ROOT, "lib"),
-            os.path.join(VAR_ROOT, "run"),
-            os.path.join(USR_ROOT, "include"),
-            os.path.join(USR_ROOT, "share")]
-    for _dir in dirs:
-        # Skip if destination is a symlink
-        if os.path.islink(os.path.join(_dir, "cms")):
-            continue
-        makedir(_dir, root, 0755)
-        _dir = os.path.join(_dir, "cms")
-        makedir(_dir, cmsuser, 0770)
-
-    print("copying Polygon testlib:")
-    path = os.path.join("cmscontrib", "polygon", "testlib.h")
-    dest_path = os.path.join(USR_ROOT, "include", "cms", "testlib.h")
-    copyfile(path, dest_path, root, 0644)
-
-    os.umask(old_umask)
-    print("done.")
-
-
-if __name__ == "__main__":
-    do_setup()
-    if "build" in sys.argv:
-        build()
-    if "install" in sys.argv:
-        install()
+from setuptools import setup, find_packages
+
+
+PACKAGE_DATA = {
+    "cms.io": [
+        os.path.join("static", "*"),
+    ],
+    "cms.server": [
+        os.path.join("static", "jq", "*.*"),
+        os.path.join("static", "sh", "*.*"),
+        os.path.join("static", "css", "*.*"),
+        os.path.join("static", "js", "*.*"),
+        os.path.join("static", "img", "*.*"),
+        os.path.join("static", "img", "mimetypes", "*.*"),
+        os.path.join("static", "*.*"),
+        os.path.join("templates", "contest", "*.*"),
+        os.path.join("templates", "admin", "*.*"),
+        os.path.join("templates", "admin", "fragments", "*.*"),
+        os.path.join("templates", "ranking", "*.*"),
+    ],
+    "cms.service": [
+        os.path.join("templates", "printing", "*.*"),
+    ],
+    "cmsranking": [
+        os.path.join("static", "img", "*.*"),
+        os.path.join("static", "lib", "*.*"),
+        os.path.join("static", "*.*"),
+    ],
+    "cmstestsuite": [
+        os.path.join("code", "*.*"),
+        os.path.join("tasks", "batch_stdio", "data", "*.*"),
+        os.path.join("tasks", "batch_fileio", "data", "*.*"),
+        os.path.join("tasks", "batch_fileio_managed", "code", "*"),
+        os.path.join("tasks", "batch_fileio_managed", "data", "*.*"),
+        os.path.join("tasks", "communication", "code", "*"),
+        os.path.join("tasks", "communication", "data", "*.*"),
+    ],
+}
+
+# Apparently, pip installs package_data with the permissions they
+# have on the source. We fix the source permissions here. (Though,
+# pip works on a copy, so no changes happen.)
```

Comment:
No problem. Anyway I think we can get rid of them: it seems that they fix the permissions directly on the file system, so it looks like it's "enough" to run them once and commit the permissions that changed (that, in my test, happened to be none).

Btw, we could avoid using package_data completely and [switch to a MANIFEST.in file](http://blog.codekills.net/2011/07/15/lies,-more-lies-and-python-packaging-documentation-on--package_data-/).